spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-pulsar/src/test/java/smoketest/pulsar/SamplePulsarApplicationTests.java

```diff
@@ -50,17 +54,40 @@ static void pulsarProperties(DynamicPropertyRegistry registry) {
 		registry.add("spring.pulsar.admin.service-url", PULSAR_CONTAINER::getHttpServiceUrl);
 	}
 
-	@Test
-	void appProducesAndConsumesSampleMessages(@Autowired SampleMessageConsumer consumer) {
-		Awaitility.await()
-			.atMost(Duration.ofMinutes(3))
-			.with()
-			.pollInterval(Duration.ofMillis(500))
-			.untilAsserted(() -> hasExpectedIds(consumer));
+	@Nested
+	@SpringBootTest
```

Comment:
While these do not use the same spring boot context, they do share the single pulsar container and this makes the test about 2x as fast.